pkg/controller/oneagent/oneagent_controller.go

```diff
@@ -245,6 +255,44 @@ func (r *ReconcileOneAgent) buildDynatraceClient(instance *dynatracev1alpha1.One
 	return dtc, err
 }
 
+func (r *ReconcileOneAgent) reconcileNodesMarkedForDeletion(reqLogger logr.Logger, dtc dtclient.Client) error {
+
+	// nodes := &corev1.NodeList(metav1.ListOptions{})
+	nodeList := &corev1.NodeList{}
+	fieldSelector := fields.SelectorFromSet(fields.Set{
+		"spec.unschedulable": "True",
+	})
+	listOps := &client.ListOptions{
```

Comment:
The OneAgent object also has a node selector that should be included here. Customers may have multiple OneAgent objects with different node selectors (e.g., production and staging, etc.)

Reply:
Node selector for selecting nodes? I dont think the nodes are controller objects, and the only relevance is them being cordoned.

Reply:
A given OneAgent object has a node selector, which says which nodes will have the OneAgent pod running. It doesn't necessarily have to be all nodes in the cluster.

You may receive the cordoning notification about a node not being monitored by the OneAgent and yet still trying to send it.